lib/rules/no-leading-uppercase-api-name.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+const { isApiDecorator } = require('../util/decorator-util');
+
+const LEADING_UPPER_CASE_REGEX = new RegExp('^[A-Z.]');
```

Comment:
Why use a regex testing ascii A-Z instead of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toUpperCase to deal with all character ranges? Please verify with @caridy that this works across locales. 

If you stick with a regex why is `.` in the regex?

Reply:
We are only interested in testing that the leading character is an upper case.